drivers/backup/portworx/portworx.go

```diff
@@ -1535,6 +1565,107 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
 	return nil
 }
 
+// CreateRuleForBackupWithMultipleApplications creates backup rule for multiple application in one rule
+func (p *portworx) CreateRuleForBackupWithMultipleApplications(ruleName string, orgID string, appList []string, prePostFlag string, ctx context.Context) (bool, string, error) {
```

Comment:
How about CreateRuleForBackupFromAppParameters?

Reply:
I just wanted to mention "MultipleApplication" in the name because as discussed we already have the method to create rule
But at present that method will create different rule for different app
In case we create backup with multiple applications, we will not be able to use rules for each application while taking backup.
Let me know if I should still change the name or suggest any other name .

Reply:
Are you not planning to take AppParameters as input?

Reply:
AppParameters is like a superset containing details for each app
eg : As of now we have postgres, cassandra and mysql  details is AppParameters
I will deploy only cassandra and mysql for backup[not postgres]. Then I will pass appList which has value of only cassandra and mysql taken from ginkgo cli.

Reply:
1) I am not sure about this
2)Why would I want to work with postgres rule If I do not have postgres deployed?

Reply:
I thought you were creating one rule for all applications. My bad. How about you take both AppParameters and AppList as inputs and create rules based on the AppList? For instance, if I want to use custom rules for a test case, then I can create new custom AppParameters and use this function; otherwise, I would need to create a similar function again.

Reply:
Agree with Krishna. @snigdha-px I would suggest to have on struct for each application and it should have its own implementation for giving pre and post rules. Please implement from that perspective.

Reply:
done